pos_tare/static/src/js/screens.js

```diff
@@ -0,0 +1,202 @@
+
+    "use strict";
+    var core = require('web.core');
+    var screens = require('point_of_sale.screens');
+    var utils = require('web.utils');
+
+    var _t = core._t;
+    var round_pr = utils.round_precision;
+    var leq_zero_qty = (ol) => ol.get_quantity() <= 0;
+
+    // This configures read action for tare barcode. A tare barcode contains a
+    // fake product ID and the weight to be subtracted from the product in the
+    // latest order line.
+    screens.ScreenWidget.include({
+        barcode_tare_action: function (code) {
+            try {
+                var order = this.pos.get_order();
+                var selected_order_line = order.get_selected_orderline();
+                var tare_weight = code.value;
+                selected_order_line.set_tare(tare_weight, true);
+            } catch (error) {
+                var title = _t("We can not apply this tare barcode.");
+                var popup = {title: title, body: error.message};
+                this.gui.show_popup('error', popup);
+            }
+        },
+        // Setup the callback action for the "weight" barcodes.
+        show: function () {
+            this._super();
+            if (this.pos.config.iface_tare_method !== 'manual') {
+                this.pos.barcode_reader.set_action_callback(
+                    'tare',
+                    _.bind(this.barcode_tare_action, this));
+            }
+        },
+    });
+
+    screens.ScaleScreenWidget.include({
+
+        // /////////////////////////////
+        // Overload Section
+        // /////////////////////////////
+
+        show: function () {
+            this.tare = 0.0;
+            this.gross_weight = 0.0;
+            this._super();
+            var self = this;
+            this.$('#input_weight_tare').keyup(function (event) {
+                self.onchange_tare(event);
+            });
+            this.$('#input_gross_weight').keyup(function (event) {
+                self.onchange_gross_weight(event);
+            });
+            if (this.pos.config.iface_gross_weight_method === 'scale') {
+                this.$('#input_weight_tare').focus();
+            } else {
+                this.pos.proxy_queue.clear();
+                this.$('#input_gross_weight').focus();
+            }
+        },
+
+        // Overload set_weight function
+        // We assume that the argument is now the gross weight
+        // we compute the net weight, depending on the tare and the gross weight
+        // then we call super, with the net weight
+        set_weight: function (gross_weight) {
+            this.gross_weight = gross_weight;
+            var net_weight = gross_weight - (this.tare || 0);
+            this.$('#container_weight_gross').text(
+                this.get_product_gross_weight_string());
+            this._super(net_weight);
+        },
+
+        order_product: function () {
+            // TODO Set a warning, if the value is incorrect;
```

Comment:
Sometimes I forget my todoes, maybe it's a similar case here :man_shrugging:

Reply:
It is!

Reply:
done.